src/session_concurrency/request_handler.cc

```diff
@@ -71,6 +71,8 @@ RequestHandler::SendRequestAndWaitForResponse(
 {
   auto payload{payload_dataset_manager_->GetPayload(dataset_index)};
 
+  PayloadJsonUtils::RemoveSessionID(payload);
+
```

Comment:
This is the place where we would take the next payload from the input dataset and add the chat history into it (the multi-turns).

So it's also a good place to strip out the session ID, which would have been present in the input dataset, but now the requirements changed so we shouldn't send the session ID with the payload.